internal/sync/syncer_related.go

```diff
@@ -70,10 +70,32 @@ func (s *ResourceSyncer) processRelatedResource(log *zap.SugaredLogger, stateSto
 		dest = local
 	}
 
-	// to find the source related object, we first need to determine its name/namespace
-	sourceKey, err := resolveResourceReference(source.object, relRes.Reference)
-	if err != nil {
-		return false, fmt.Errorf("failed to determine related object's source key: %w", err)
+	var sourceKey *ctrlruntimeclient.ObjectKey
+	switch {
+	case relRes.Reference != nil:
+		// to find the source related object, we first need to determine its name/namespace
+		sourceKey, err = resolveResourceReference(source.object, *relRes.Reference)
+		if err != nil {
+			return false, fmt.Errorf("failed to determine related object's source key: %w", err)
+		}
+	case relRes.LabelSelector != nil:
+		// if no reference is given, we can use a label selector to find the source object
+		sourceObjs := &unstructured.UnstructuredList{}
+		sourceObjs.SetAPIVersion("v1")
+		sourceObjs.SetKind(relRes.Kind)
+
+		// TODO: would need to handle replacer here as well to select right object.
```

Comment:
This is where it get tricky. You want to be able to use projections, and I think only place where we can get it is from "owning object" to which this resource is related. So this might need some wiring in...